src/components/admin/semesters/SemesterCreateDialog.tsx

```diff
@@ -0,0 +1,187 @@
+import React from "react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useForm, type SubmitHandler } from "react-hook-form";
+import { z } from "zod";
+
+import { TextInput } from "@/components/TextInput";
+import { useToast } from "@/components/ui/use-toast";
+import {
+  DialogCard,
+  DialogCardFooter,
+  useDialogContext,
+} from "@/components/ui/utils/DialogUtils";
+import { compareDate, formatDateInISO, validateDate } from "@/lib/utils";
+
+//Schema
+const formSchema = z
+  .object({
+    name: z.string().min(1, "Field is required"),
+    startDate: z
+      .string()
+      .min(1, "Field is required")
+      .refine(validateDate, "Invalid date"),
+    endDate: z
+      .string()
+      .min(1, "Field is required")
+      .refine(validateDate, "Invalid date"),
+    breakStart: z
+      .string()
+      .min(1, "Field is required")
+      .refine(validateDate, "Invalid date"),
+    breakEnd: z
+      .string()
+      .min(1, "Field is required")
+      .refine(validateDate, "Invalid date"),
+  })
+  .refine((data) => compareDate(data.startDate, data.breakStart) < 0, {
+    message: "Start date must be before than break start date",
+    path: ["startDate"],
+  })
+  .refine((data) => compareDate(data.breakStart, data.breakEnd) < 0, {
+    message: "Break start date start must be before than break end date",
+    path: ["breakStart"],
+  })
+  .refine((data) => compareDate(data.breakEnd, data.endDate) < 0, {
+    message: "Break end date must be before than end date",
+    path: ["breakEnd"],
+  });
+
+export const SemesterCreateDialog = () => {
+  //Context
+  const { handleClose: closeDialog } = useDialogContext();
+
+  // Hook-forms
+  const {
+    register,
+    handleSubmit,
+    reset,
+    setError,
+    formState: { errors },
+  } = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+  });
+  const { toast } = useToast();
+
+  // React-query
+  const queryClient = useQueryClient();
+  const mutation = useMutation({
+    mutationFn: async (body: BodyInit) => {
+      const response = await fetch("/api/semesters", {
+        method: "POST",
+        body,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+
+      if (!response.ok) {
+        await response.text().then((text) => {
+          // todo: handle this better somehow
+          if (text == "This name already exists, please pick another") {
+            setError("name", { message: text });
+          }
+          throw new Error(text || "An has error occurred");
+        });
```

Comment:
you could consider attaching something to the error message to detect that the error is due to a duplicate semester name

e.g. return a json with errorCode field that identifies the type of error

i want to eventually overhaul how errors are returned from the backend to make things more detailed too.